packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapController.m

```diff
@@ -119,24 +125,20 @@ - (instancetype)initWithFrame:(CGRect)frame
     if ([tileOverlaysToAdd isKindOfClass:[NSArray class]]) {
       [_tileOverlaysController addTileOverlays:tileOverlaysToAdd];
     }
+
+    [_mapView addObserver:self forKeyPath:@"frame" options:0 context:nil];
   }
   return self;
 }
 
 - (UIView *)view {
-  [_mapView addObserver:self forKeyPath:@"frame" options:0 context:nil];
   return _mapView;
 }
 
 - (void)observeValueForKeyPath:(NSString *)keyPath
                       ofObject:(id)object
                         change:(NSDictionary *)change
                        context:(void *)context {
-  if (_cameraDidInitialSetup) {
-    // We only observe the frame for initial setup.
```

Comment:
Rather than remove this comment, it would be good to move it down to just above the remaining call to `removeObserver:forKeyPath:`.